### PR TITLE
Remove static initializer race condition

### PR DESCRIPTION
**Pull request by MarkZuber**
From issue [629] we have a static race condition related to PlatformPlugin and CorePlatformInformationBase static initializers.  This removes the statics there and forces the core classes to hit the module initializer directly for the remaining statics we're on a path to removing over time.

### CODE REVIEW

File: msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs [hunk 1]
```diff
@@ -93,10 +102,19 @@ public abstract partial class ClientApplicationBase
 
             RequestContext requestContext = new RequestContext(new MsalLogger(Guid.Empty, null));
 
+            var platformInformation = new PlatformInformation();
+
```
Comment (jennyf19) [inline msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs]: is there a reason why you can't use the PlatformInformation from line 91? #Resolved

File: msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs [hunk 2]
```diff
@@ -93,10 +102,19 @@ public abstract partial class ClientApplicationBase
 
             RequestContext requestContext = new RequestContext(new MsalLogger(Guid.Empty, null));
 
+            var platformInformation = new PlatformInformation();
```
Comment (bgavrilMS) [inline msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs]: Can't you use the instance created at line 91?  #Closed

**Review by bgavrilMS — APPROVED** [review]

File: msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs (hunk 1 above)
Comment (MarkZuber) [reply msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs]: fixed

File: msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs (hunk 2 above)
Comment (MarkZuber) [reply msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs]: fixed #Closed

**Review by jennyf19 — APPROVED** [review]
:shipit:

File: msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs [hunk 3]
```diff
@@ -32,6 +32,7 @@
 using Microsoft.Identity.Client.Internal;
 using Microsoft.Identity.Client.Internal.Requests;
 using System.Linq;
+using System.Reflection;
```
Comment (henrik-me) [inline msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs:35]: >ing System.Reflection; [](start = 2, length = 22)

why is this needed?

Comment (MarkZuber) [reply msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs:35]: We moved AssemblyInformationalVersionAttribute in here (with the GetAssemblyInformationalVersion method) and that's in the system.reflection namespace.

File: msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs [hunk 4]
```diff
@@ -95,8 +104,15 @@ public abstract partial class ClientApplicationBase
...
+                PlatformInformation.GetProductName(), MsalIdHelper.GetMsalVersion(),
+                PlatformInformation.GetAssemblyFileVersionAttribute(), GetAssemblyInformationalVersion()));
+        }
+
+        private static string GetAssemblyInformationalVersion()
+        {
+            AssemblyInformationalVersionAttribute attribute =
+                typeof(ClientApplicationBase).GetTypeInfo().Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>();
+            return (attribute != null) ? attribute.InformationalVersion : string.Empty;
         }
```
Comment (henrik-me) [inline msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs:116]: what about moving this to an external helper class? perhaps the platforminformation can call that one... (see also my comment on referencing system.reflection)

**Comment by henrik-me** [comment]
Makes sense to pass the platform information along and putting it on the application object. Not sure why it needs to be new'ed up in some places in the code. Can you pls. elaborate on that?

File: msal/tests/dev apps/XForms/XForms/CachePage.xaml.cs [hunk 5]
```diff
@@ -96,7 +97,7 @@ private async Task OnClearClickedAsync(object sender, EventArgs e)
         {
             var tokenCache = App.MsalPublicClient.UserTokenCache;
             var users = await tokenCache.GetAccountsAsync
```
Comment (henrik-me) [inline msal/tests/dev apps/XForms/XForms/CachePage.xaml.cs]: >await tokenCache.GetAccountsAsync [](start = 24, length = 33)

can we grab this directly from the PCA? #Closed

Comment (MarkZuber) [reply msal/tests/dev apps/XForms/XForms/CachePage.xaml.cs]: fixed #Closed

**Comment by henrik-me** [comment]
related issue: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/629

**Review by henrik-me — APPROVED** [review]
:shipit:
